plugins/main/ipvlan/ipvlan.go

```diff
@@ -191,9 +191,20 @@ func cmdDel(args *skel.CmdArgs) error {
 		return nil
 	}
 
-	return ns.WithNetNSPath(args.Netns, func(_ ns.NetNS) error {
-		return ip.DelLinkByName(args.IfName)
+	// Check whether the device exists before trying to delete it
+	devErr := ns.WithNetNSPath(args.Netns, func(_ ns.NetNS) error {
+		_, err := netlink.LinkByName(args.IfName)
+		return err
 	})
+
+	// Only try to delete the device if it could be found in the first place.
+	if devErr == nil {
```

Comment:
just a silly question is there any reason to not bail out if `devErr != nil`  rather than continuing if `devErr == nil`. maybe with a comment about the device not existing/not being found is not an exceptional error which should be returned